### PR TITLE
Action controller overview

### PR DESCRIPTION
**Pull request by Krutik31**

### CODE REVIEW

File: app/controllers/cars_controller.rb [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def edit
+    @car = Car.find(params[:id])
+  end
+
+  def update
+    @car = Car.find(params[:id])
```
Comment (purvak-simform) [inline app/controllers/cars_controller.rb]: Use Filter for it

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+class UsersController < ApplicationController
+  http_basic_authenticate_with name: 'user', password: '123'
+
+  def index; end
+
+  def create
+    @user = User.where(username: params[:username]).where(password: params[:password]).first
```
Comment (purvak-simform) [inline app/controllers/users_controller.rb]: merge it in one query
User.where(username: "", password: "")

File: app/views/cars/_logout.html.erb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<div class="w-50 bg-dark mx-auto text-light">
+  <%=flash[:notice]%>
+</div>
+<div class="row">
+  <div class="col-6 text-left">
+    <h4>Hello <%=cookies[:user_name].titleize%> (<%= link_to "download pdf", download_pdf_path %>) </h4>
+  </div>
+  <div class="col-6">
+    <%= link_to "Logout", user_path(current_user.id), data: { turbo_method: :delete }, class:"btn btn-danger float-right" %>
```
Comment (purvak-simform) [inline app/views/cars/_logout.html.erb]: is it required to send current_user.id ?

File: app/views/cars/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+  <h2 class="text-center">Add New Car</h2>
+
+  <div class="row">
+    <div class="col-3">
+    </div>
+    <div class="col-6">
+      <%= render partial: "logout" %>
+      <% if @car.errors.any? %>
+        <hr>
```
Comment (purvak-simform) [inline app/views/cars/new.html.erb]: For same code create partial.Here for error code snippet

**Review by purvak-simform — CHANGES_REQUESTED** [review]

File: app/views/cars/_logout.html.erb (hunk 3 above)
Comment (Krutik31) [reply app/views/cars/_logout.html.erb]: No, its not required.
let me refactor it.

**Review by purvak-simform — APPROVED** [review]

**Pull request merged**